packages/itwinui-react/src/core/ToggleSwitch/ToggleSwitch.tsx

```diff
@@ -53,10 +61,15 @@ export type ToggleSwitchProps = {
  */
 export const ToggleSwitch = React.forwardRef(
   (props: ToggleSwitchProps, ref: React.RefObject<HTMLInputElement>) => {
+    let icon: JSX.Element | undefined;
+    if (props.size !== 'small') {
+      icon = props.icon;
+      props = { ...props };
+      delete props.icon;
+    }
```

Comment:
i wonder if we should make a utility function to hide this mess

Reply:
Maybe we should. There are a lot of places where we do this. Maybe in next PR to not make unrelated changes in this one.